### ISSUE TITLE
[v14.x] By default Jessie won't allow SSH password login (forces key-pairs)

### ISSUE THREAD

**Issue by JedMeister** [turnkeylinux/tracker#324]
I'm labelling this as a bug as it will need to be dealt with for v14.0 release.

In Jessie OpenSSH server doesn't allow password login by default and I think that we need to change that...

Following discussion with @alonswartz here are a few points:
* It should go in `common/conf/turnkey.d/sshd`
* There are a couple of ways to go about this:
  1. Base off codename
  * Base off current configuration

1. is easier, solves a specific change in specific version. But, we'll need to update that code every release. Not the end of the world..

2. might be harder, but more generic. We could check the current state of the configuration and update it to the end-state as required by the policy. Ie, check if root_pass auth is enabled, if not, enable it.

**Comment by JedMeister** [comment]
FWIW originally I suggested 1 (from above). Something like this:
````
if [ -z $CODENAME ]; then CODENAME=$(lsb_release -s -c) fi

if [ "$CODENAME" = "jessie" ]; then sed -i 's/PermitRootLogin without-password/PermitRootLogin yes/' /etc/ssh/sshd_config fi
````

But considering that 2 is a better option really we'll want to refactor that...

**Comment by alon** [comment]
@JedMeister I think you got the wrong user in the issue comment (@-alon is me).

**Comment by JedMeister** [comment]
orry @alon; fixed!

**Comment by alonswartz** [comment]
Closed with commit: turnkeylinux/common@0c595677
